src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -249,124 +249,128 @@ public RTPEncodingDesc(
+     * newerFrame, assuming olderFrame predates/is older than newerFrame.
+     * Depending on the relationship of olderFrame and newerFrame, and what
+     * we know about olderFrame and newerFrame, we may be able to deduce
+     * the last expected sequence number for olderFrame and/or the first
+     * expected sequence number of newerFrame.
      *
-     * @param a the {@link FrameDesc} that comes before b.
-     * @param b the {@link FrameDesc} that comes after a.
+     * @param olderFrame the {@link FrameDesc} that comes before newerFrame.
+     * @param newerFrame the {@link FrameDesc} that comes after olderFrame.
      */
-    private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
+    private static void applyFrameBoundsHeuristics(
+            FrameDesc olderFrame,
+            FrameDesc newerFrame)
     {
-        int aLastSeqNum = a.getEnd(), bFirstSeqNum = b.getStart();
-        if (aLastSeqNum != -1 && bFirstSeqNum != -1)
+        if (olderFrame.lastSequenceNumberKnown() && newerFrame.lastSequenceNumberKnown())
         {
-            // No need for heuristics.
+            // We already know the last sequence number of olderFrame and the first
+            // sequence number of newerFrame, no need for further heuristics.
             return;
         }
-
-        long tsDiff = (b.getTimestamp() - a.getTimestamp()) & 0xFFFFFFFFL;
-        if (tsDiff > (1L << 30) && tsDiff < (-(1L << 30) & 0xFFFFFFFFL))
+        if (!TimestampUtils.isNewerTimestamp(
+                newerFrame.getTimestamp(), olderFrame.getTimestamp()))
         {
-            // the distance (mod 2^32) between the two timestamps needs to be
-            // less than half the timestamp space.
+            // newerFrame isn't newer than olderFrame, bail
             return;
         }
-        else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
-        {
-            logger.warn("Frames that are out of order detected.");
-        }
-        else
+        int lowestSeenSeqNumOfNewerFrame = newerFrame.getMinSeen();
+        int highestSeenSeqNumOfOlderFrame = olderFrame.getMaxSeen();
+        int seqNumDiff =
+                RTPUtils.sequenceNumberDiff(lowestSeenSeqNumOfNewerFrame, highestSeenSeqNumOfOlderFrame);
+
+        boolean guessed = false;
+
+        // We have some codecs (VPX) for which we can inspect the payload
+        // to deduce the start/end of olderFrame frame. This has an impact on the
+        // distances between the frames that allow us to guess frame
+        // boundaries.
+        //FIXME(brian): why does this allow us to infer anything extra from the sequence number
+        // differences?  we don't leverage frame boundary markings in any way when we deduce
+        // first/last sequence numbers.  does it just mean that these frames would've already
+        // (conclusively) had the start/end seq numbers set?  if so, we wouldn't need to
+        // call this method at all for those frames, right?
```

Comment:
> is it because if packet 0 were marked as the end, we would have already known it at this point and it would've be set by now? and, similarly, if packet 3 was the start we would have already known it at this point and it would've been set by now?

Yes, that's it!